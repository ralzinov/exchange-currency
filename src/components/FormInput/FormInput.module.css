.host {
    padding: 12px;
    border-radius: 4px;
    border: 1px solid #ccc;
    font-size: 16px;
    outline: none;
    margin: 0;
}

.host:focus {
    border-color: #2684FF;
    border-width: 2px;
    margin: -1px;
}

.host.invalid {
    border-color: #ff6b6b;
    border-width: 2px;
    margin: -1px;
}
