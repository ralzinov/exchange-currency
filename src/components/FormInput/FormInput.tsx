import * as React from 'react';
import classNames from 'classnames';
import styles from './FormInput.module.css';
import {WrappedFieldProps} from 'redux-form';
import {ValidationMessage} from '../ValidationMessage';

export interface IFormInputProps {
    className?: string;
    placeholder?: string;
    disabled?: boolean;
    hidden?: boolean;
}

export const FormInput: React.FC<WrappedFieldProps & IFormInputProps> = (props) => {
    const {error, touched, invalid, dirty} = props.meta;
    const className = classNames(
        invalid && touched && styles.invalid,
        props.className,
        styles.input
    );

    return (
        <div className={styles.host}>
            <input
                placeholder={props.placeholder || ''}
                className={className}
                disabled={props.disabled}
                hidden={props.hidden}
                {...props.input}
            />
            <ValidationMessage className={styles.validationMessage}>
                {touched || dirty ? error : ''}
            </ValidationMessage>
        </div>
    );
};
