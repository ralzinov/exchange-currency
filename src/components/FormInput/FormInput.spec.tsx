import React from 'react';
import ReactDOM from 'react-dom';
import {FormInput, IFormInputProps} from './FormInput';
import {create} from 'react-test-renderer';
import {WrappedFieldProps} from 'redux-form';

it('renders without crashing', () => {
    const metaProps: PartialDeep<WrappedFieldProps> = {
        meta: {
            error: 'Error message',
            touched: true,
            invalid: true,
            dirty: true
        }
    };

    const div = document.createElement('div');
    ReactDOM.render(<FormInput {...metaProps as WrappedFieldProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
});

it('should render with props in invalid state', () => {
    const props: PartialDeep<WrappedFieldProps & IFormInputProps> = {
        placeholder: 'Some placeholder',
        className: 'input',
        disabled: false,
        hidden: false,
        meta: {
            error: 'Error message',
            touched: true,
            invalid: true,
            dirty: true
        }
    };
    const component = create(
        <FormInput {...props as (WrappedFieldProps & IFormInputProps)}/>
    );
    expect(component.toJSON()).toMatchSnapshot();
});

it('should not render error state if untouched and pristine', () => {
    const props: PartialDeep<WrappedFieldProps & IFormInputProps> = {
        meta: {
            error: 'Error message',
            touched: false,
            invalid: true,
            dirty: false
        }
    };
    const component = create(
        <FormInput {...props as (WrappedFieldProps & IFormInputProps)}/>
    );
    expect(component.toJSON()).toMatchSnapshot();
});
