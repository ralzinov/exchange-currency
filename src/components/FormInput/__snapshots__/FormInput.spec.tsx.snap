// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not render error state if untouched and pristine 1`] = `
<div
  className="host"
>
  <input
    className="input"
    placeholder=""
  />
  <div
    className="host validationMessage"
  >
    
  </div>
</div>
`;

exports[`should render with props in invalid state 1`] = `
<div
  className="host"
>
  <input
    className="invalid input input"
    disabled={false}
    hidden={false}
    placeholder="Some placeholder"
  />
  <div
    className="host validationMessage"
  >
    Error message
  </div>
</div>
`;
